docs/native-modules.md

```diff
@@ -293,17 +293,25 @@ Then we define constants, and it's as easy as creating a public field and giving
 
 It's just as easy to add custom methods, by attributing a public method with `REACT_METHOD`. In FancyMath we have one method, `add`, which takes two doubles and returns their sum. Again, we've specified the optional `name` argument in the `REACT_METHOD` macro-attribute so in JS we call `add` instead of `Add`.
 
-Methods that return more complex types (like `std::string`) can be implemented by returning void and accepting a bool and a `ReactPromise<JSValue>`:
+Native modules do not have the ability to check that the parameter types and the number of parameters match between what's called from JavaScript and what the native code accepts. However, the framework will validate that the number of promises-like parameters match: if the JavaScript API is async, it will expect that there is one "promise-like" parameter in the native method implementation signature.
+
+A "promise-like" parameter is either:
+- `React::ReactPromise<T>`
+- a callback function or functor.
+
+See [Using Asynchronous Windows APIs](native-modules-async.md).
+
+Here is an example of an async method that returns a string. 
 
 ```cpp
     REACT_METHOD(GetString, L"getString");
-    void GetString(bool error, React::ReactPromise<React::JSValue>&& result) noexcept
+    void GetString(React::ReactPromise<std::string>&& result) noexcept
     {
       if (error) {

```

Comment:
>error [](start = 10, length = 5)

Should we make the example more realistic? This variable is undefined. E.g. we can show this method calling some WinRT API.

Reply:
Add more realistic examples of native modules calling WinRT APIs
Including pointing to our sample native modules and possibly community modules

Reply:
Filed 288